### PR TITLE
Improve C++ tests

### PR DESCRIPTION
**Pull request by rafael1193**
Use Boost.Test library to improve how the C++ tests are handled

### CODE REVIEW

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -20,10 +20,17 @@ set(SOURCE_FILES src/ext/dubins.cpp src/ext/dubins.h src/ext/json.hpp src/core/s
...
-add_executable(tests ${SOURCE_FILES} src/test/core/test_reversible_updates.h src/test/test_dubins.h
-        src/test/test_position_manipulation.h src/test/main_tests.cpp)
-
 pybind11_add_module(uav_planning ${SOURCE_FILES} src/python_interface.cpp)
 
 set(PLANNING_MODULE_NAME "uav_planning.cpython-35m-x86_64-linux-gnu.so")
 set(PLANNING_MODULE "${CMAKE_CURRENT_BINARY_DIR}/${PLANNING_MODULE_NAME}" PARENT_SCOPE)
+
+
+find_package(Boost COMPONENTS unit_test_framework REQUIRED)
```
Comment (arbimo) [inline cpp/CMakeLists.txt]: Is it possible to only require boost for the "tests" target. I know this is usually trivial to do in java related build tools but I don't know about CMake.
If it is not very easy, then just leave it like this.

**Review by arbimo — COMMENTED** [review]
Very good idea to have well defined C++ tests.

Only thing missing is to add the boost dependency to the docker environment.

**Comment by arbimo** [comment]
Also it would be good to have tests run automatically on each commit to master which can be done with Travis.
The only trick would be to get travis to run tests in the docker environment. I suspect it is fairly easy but I have never tried it.

My experience is that when tests are run automatically, we tend to write tests more often =)

**Review by arbimo — COMMENTED** [review]
To improve coverage and make those more likely to catch bugs, a simple approach is to generate many random test cases (with the same seed for reproducibility).

File: cpp/CMakeLists.txt (hunk 1 above)
Comment (rafael1193) [reply cpp/CMakeLists.txt]: I am a complete noob in CMake so I haven't found yet how to do it. However I will try to find out a solution.

**Comment by rafael1193** [comment]
Travis CI support is considered in #5 

I think the branch can be merged in the current state. Increasing the number of tests and make them random can be treated in the future.

**Comment by arbimo** [comment]
Yes, feel free to merge :)

Le 26 oct. 2017 4:59 PM, "Rafael Bailón" <notifications@github.com> a
écrit :

Travis CI support is considered in #5
<https://github.com/fire-rs-laas/fire-rs-saop/pull/5>

I think the branch can be merged in the current state. Increasing the
number of tests and make them random can be treated in the future.

—
You are receiving this because your review was requested.

Reply to this email directly, view it on GitHub
<https://github.com/fire-rs-laas/fire-rs-saop/pull/4#issuecomment-339694274>,
or mute the thread

**Pull request merged**